### PR TITLE
Adding helpers for zcl cli commands such that the right content is generated for Zigbee and other cleanup.

### PR DESCRIPTION
**Pull request by brdandu**
- Increasing the feature level pointer for apacka.json
- Checking for lowercase when querying bitmaps and enums since the atomics are in uppper case under certain occasions for eg BITMAP8 instead of bitmap8
- Getting rid of the cli helpers in helper-zcl.js and using the new data_type table after the schema change instead of the old logic. Also moving the cli helpers int helper-zigbee-zcl since it is specific to zigbee 3.0 only
- Adding helper-zigbee-zcl.js to template-engine such that the helpers are included
- Adding the new zcl cli helper in zap-cli.zapt such that testing of the new helper is covered along with the old one.
- JIRA: ZAPP-802

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: src-electron/db/query-bitmap.js [hunk 1]
```diff
@@ -87,8 +87,8 @@ SELECT
   BITMAP.SIZE AS SIZE
 FROM BITMAP
 INNER JOIN DATA_TYPE ON BITMAP.BITMAP_ID = DATA_TYPE.DATA_TYPE_ID
-WHERE DATA_TYPE.NAME = ? AND DATA_TYPE.PACKAGE_REF = ?`,
-      [name, packageId]
+WHERE (DATA_TYPE.NAME = ? OR DATA_TYPE.NAME = ?) AND DATA_TYPE.PACKAGE_REF = ?`,
+      [name, name.toLowerCase(), packageId]
```
Comment (jingteng25742) [inline src-electron/db/query-bitmap.js:91]: Are both the input arg 'name' and the column attribute DATA_TYPE.NAME fields case-insensitive?
If 'name' is only lower cased but nam DATA_TYPE.NAME has some weird capitalization, they might not necessarily match.
Using `COLLATE NOCASE` in the SQL query would avoid that issue.

File: src-electron/db/query-enum.js [hunk 2]
```diff
@@ -244,9 +244,9 @@ INNER JOIN
 ON
   ENUM.ENUM_ID = DATA_TYPE.DATA_TYPE_ID
 WHERE
-  NAME = ? AND PACKAGE_REF = ?
+  (DATA_TYPE.NAME = ? OR DATA_TYPE.NAME = ?)AND PACKAGE_REF = ?
 ORDER BY NAME`,
-      [name, packageId]
+      [name, name.toLowerCase(), packageId]
```
Comment (jingteng25742) [inline src-electron/db/query-enum.js:249]: same as earlier comment about case-insensitiveness.

File: src-electron/generator/helper-zcl.js [hunk 3]
```diff
@@ -2614,8 +2466,6 @@ exports.if_manufacturing_specific_cluster = dep(
   if_manufacturing_specific_cluster,
   { to: 'if_mfg_specific_cluster' }
 )
-exports.zcl_command_argument_type_to_cli_data_type =
-  zcl_command_argument_type_to_cli_data_type
```
Comment (jingteng25742) [inline src-electron/generator/helper-zcl.js:2618]: do we need to maintain backward compatibility for the zcl_command_argument_type_to_cli_data_type api under helper-zcl.js?

File: src-electron/db/query-bitmap.js (hunk 1 above)
Comment (brdandu) [reply src-electron/db/query-bitmap.js:91]: So I thought about this. I used like to overcome this at one point but didn't want to promote this so just kept it to either small/uppercase. None of the other variations.

File: src-electron/generator/helper-zcl.js (hunk 3 above)
Comment (brdandu) [reply src-electron/generator/helper-zcl.js:2618]: This helper is still present and generates the correct content in helper-zigbee-zcl.js

**Review by jingteng25742 — APPROVED** [review]

**Pull request merged**